### ISSUE TITLE
refs in items of array do not work at root level

### ISSUE THREAD

**Issue by AndyOGo** [adobe/jsonschema2md#72]
## What did you do

I define an array and used `$ref`s for all it's items.

## What did you expect to happen

I expected a nice list of all linked references.

## What happened

Nothing displayed

## What's your environment

* Operating System: Mac OS X El Capitan
* node.js version: `v9.4.0`

## Do you have example files:

Sure, I even have a [PR](https://github.com/axa-ch/aletheia/pull/93) of a public repo.

For this schema
```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "$id": "http://www.axa.ch/schemas/types/children.json",

  "type": "array",
  "title": "Children",
  "description": "Components rendered within a Component.",

  "additionalItems": false,
  "items": [
    { "$ref": "http://www.axa.ch/schemas/components/Autocomplete.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/BulletRadioButtons.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Button.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Checkbox.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Datepicker.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Dropdown.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Error.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Icon.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Link.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/List.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/ListItem.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Map.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Markdown.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/ModalDialog.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/MultipleSelectList.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/MultipleSelectListItem.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/ProcessEnding.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/SegmentedRadioButtons.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/SelectList.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/SelectListItem.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Textarea.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Textfield.json#" },
    { "$ref": "http://www.axa.ch/schemas/components/Typo.json#" }
  ]
}
```

I'm getting following Markdown

```markdown

# Children Schema

```
http://www.axa.ch/schemas/types/children.json
```

Components rendered within a Component.

| Abstract | Extensible | Status | Identifiable | Custom Properties | Additional Properties | Defined In |
|----------|------------|--------|--------------|-------------------|-----------------------|------------|
| Can be instantiated | No | Experimental | No | Forbidden | Permitted | [types/children.json](types/children.json) |
```

**Comment by AndyOGo** [comment]
guess this is releated to #73

**Comment by Relequestual** [comment]
The schema you have created will not do what you expect in terms of JSON Schema.
You need to wrap your array of references inside a `oneOf`.

With `items`: 
- If the value is a object, it must be a json schema, and every item in the array is checked against the schema.
- If the value is an array (which is in your example), then each schema object in that array is applied to the instance array at the same position in the array.

The result is, what you have here would check that, the first item in the array is Autocomplete, and the second item in the array is BulletRadioButtons, etc.

I'm pretty sure that's not what you intended.

(Having the correct JSON Schema here doesn't mean it will fix #73 though. I might get to work on that... We shall see. =] )

**Comment by AndyOGo** [comment]
@Relequestual 
You absolutely right about correct `type:array` `items` syntax.

Nevertheless, this issue is about the correct validation of the root object.
